### PR TITLE
Adds GitHub Actions unit-test workflow

### LINKED ISSUE #4: Setup CI
- [x] Create Github Action to run jest test runner on branch push

### PR DESCRIPTION
**Pull request by qafui**
# Summary
Runs unit tests with a GitHub action

Closes #4

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: 'Run Unit Tests'
+
+on: 'push'
+
+jobs:
+  test:
+    runs-on: 'ubuntu-latest'
```
Comment (jhodapp) [inline .github/workflows/unit-tests.yml]:
```suggestion
    runs-on: ubuntu-22.04
```

Comment (jhodapp) [reply .github/workflows/unit-tests.yml]: We use this specific LTS version of Ubuntu for the `ambi-rs` repo, so I think it's helpful to use the same version across the project.

**Review by jhodapp — CHANGES_REQUESTED** [review]
Just a minor suggested change, looks good David, thank you!

**Comment by qafui** [comment]
> Just a minor suggested change, looks good David, thank you!

Thank you! Ready to merge!

**Review by jhodapp — APPROVED** [review]
Looks fantastic, thank you David!

**Pull request merged**